src/ff/boolean.rs

```diff
@@ -14,6 +16,14 @@ impl Block for bool {
 #[derive(Clone, Copy, PartialEq, Debug)]
 pub struct Boolean(bool);
 
+impl ExtendableField for Boolean {
+    type ExtendedField = Gf32Bit;
+
+    fn to_extended(&self) -> Self::ExtendedField {
+        Gf32Bit::try_from(self.as_u128()).unwrap()
+    }
+}
+
 ///trait for secret sharing
```

Comment:
Not your problem, but this doc comment is not very good.  Do comments on impl even have any effect?